executable/detect.go

```diff
@@ -65,9 +65,15 @@ func (d Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error
 		return libcnb.DetectResult{}, fmt.Errorf("unable to read manifest in %s\n%w", context.Application.Path, err)
 	}
 
+	jarGlob, _ := cr.Resolve("BP_EXECUTABLE_JAR_LOCATION")
+	_, props, _ := findExecutableJAR(context.Application.Path, jarGlob)
+
 	if _, ok := m.Get("Main-Class"); ok {
 		d.Logger.Info("PASSED: 'Main-Class' manifest attribute found")
 		result.Plans[0].Provides = append(result.Plans[0].Provides, libcnb.BuildPlanProvide{Name: PlanEntryJVMApplicationPackage})
+	} else if _, ok := props.Get("Main-Class"); ok {
```

Comment:
Can we check `jarPath` (1 return object) rather than the property? Because the function `findExecutableJAR` will only find jars with that property anyway.

Reply:
This wouldn't be needed after the change proposed by Pavel.